libbeat/common/uuid.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"crypto/rand"
+	"fmt"
+)
+
+// GenerateUUID is used to generate a random UUID
+// This was copied from https://github.com/hashicorp/uuid/blob/master/uuid.go
```

Comment:
when using UUID-4 you'd want a really good seed with enough entropy + pseudo random number generate to not increase chance of collisions. If you can not fully guarantee this, UUID 5 (basically a hash value) is a viable option.